app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,48 @@
+class RecipesController < ApplicationController
+  before_action :authenticate_user!, except: %i[public_recipe index]
+  load_and_authorize_resource
+
+  def index
+    @user = User.find(current_user.id)
+    @recipes = @user.recipes
+  end
```

Comment:
- [ ] In a Rails application using Devise, the `current_user` method is provided by Devise and is typically available once a user signs in. It represents the currently authenticated user. Since current_user is already available and set correctly in your application, then the line `@user = User.find(current_user.id)` may indeed be unnecessary. Since `current_user ` already holds the authenticated user object, you can directly access its associated recipes using `current_user.recipes `instead of querying the database again with `User.find(current_user.id)`.